### PR TITLE
Changed entities Order, Product and CreatedProduct

### PR DESCRIPTION
**Pull request by vyskocilpavel**
* Changed Table name in Order Entity
* Removed JoinColumn and Lazy fetch type from CreatedProduct Entity
* Added attribute createdProducts and methods: getProductCategories, getCreatedProducts, addCreatedProduct and removeCreatedProduct
* Added methods:getSubmittedOrders, addSubmittedOrder and removeSubmittedOrder
* Changed Enities to return unmodifiableSet

### CODE REVIEW

File: TACOS-persistence/src/main/java/cz/fi/muni/TACOS/persistence/entity/User.java [hunk 1]
```diff
@@ -111,6 +111,20 @@ public void setRole(UserRole role) {
         this.role = role;
     }
 
+    public Set<Order> getSubmittedOrders() {
+        return submittedOrders;
```
Comment (balcirakpeter) [inline TACOS-persistence/src/main/java/cz/fi/muni/TACOS/persistence/entity/User.java]: You forgot to use unmodifiableSet method.

File: TACOS-persistence/src/main/java/cz/fi/muni/TACOS/persistence/entity/CreatedProduct.java [hunk 2]
```diff
@@ -42,13 +38,11 @@
     private String description;
 
     @NotNull
-    @ManyToOne(fetch= FetchType.LAZY)
-    @JoinColumn(name="order_id")
+    @ManyToOne
```
Comment (balcirakpeter) [inline TACOS-persistence/src/main/java/cz/fi/muni/TACOS/persistence/entity/CreatedProduct.java:41]: Please add @Column(nullable = false)

File: TACOS-persistence/src/main/java/cz/fi/muni/TACOS/persistence/entity/CreatedProduct.java [hunk 3]
```diff
@@ -42,13 +38,11 @@
...
     private Order order;
 
     @NotNull
-    @ManyToOne(fetch= FetchType.LAZY)
-    @JoinColumn(name="product_id")
+    @ManyToOne
```
Comment (balcirakpeter) [inline TACOS-persistence/src/main/java/cz/fi/muni/TACOS/persistence/entity/CreatedProduct.java:45]: Please add @Column(nullable = false)

**Review by balcirakpeter — CHANGES_REQUESTED** [review]

File: TACOS-persistence/src/main/java/cz/fi/muni/TACOS/persistence/entity/CreatedProduct.java (hunk 2 above)
Comment (vyskocilpavel) [reply TACOS-persistence/src/main/java/cz/fi/muni/TACOS/persistence/entity/CreatedProduct.java:41]: @Column(s) not allowed on a @ManyToOne property:

File: TACOS-persistence/src/main/java/cz/fi/muni/TACOS/persistence/entity/CreatedProduct.java (hunk 3 above)
Comment (vyskocilpavel) [reply TACOS-persistence/src/main/java/cz/fi/muni/TACOS/persistence/entity/CreatedProduct.java:45]: @Column(s) not allowed on a @ManyToOne property:

**Review by balcirakpeter — APPROVED** [review]

**Pull request merged**